components/js/m-chart-chartjs-admin.js

```diff
@@ -0,0 +1,133 @@
+var m_chart_chartjs_admin = {
+	image_step: 1
+};
+
+(function( $ ) {
+	'use strict';
+
+	// Start things up
+	m_chart_chartjs_admin.init = function() {
+		// Hide the subtitle and $y_min_value fields
+		m_chart_admin.$y_min_value.hide();
+		m_chart_admin.$subtitle_input.hide();
+
+		// Only show fields/inputs that are appropriate for the current chart type
+		var $chart_type_select = $( document.getElementById( 'm-chart-type' ) );
+		$chart_type_select.on( 'load, change', this.handle_chart_type );
+		$chart_type_select.trigger( 'change' );
+
+		// Watch for a new chart to be built
+		if ( 'default' === m_chart_admin.performance && 'yes' === m_chart_admin.image_support ) {
+			$( '.m-chart' ).on( 'render_done', this.generate_image_from_chart );
+		}
+
+		$( '.m-chart' ).on( 'chart_args_success', this.refresh_chart );
+	};
+
+	// Handle chart type input changes so the settings UI only reflects appropriate options
+	m_chart_chartjs_admin.handle_chart_type = function( event ) {
+		var chart_type        = $( this ).attr( 'value' );
+		var $chart_meta_box   = $( document.getElementById( 'm-chart' ) );
+		var $spreadsheet_tabs = $( document.getElementById( 'hands-on-table-sheet-tabs' ) );
+
+		// Show everything before hiding the options we don't want
+		$chart_meta_box.find( '.row, .shared' ).removeClass( 'hide' );
+		$chart_meta_box.find( '.row.two' ).addClass( 'show-shared' );
+
+		// Chart.js will never need the tabs
+		$spreadsheet_tabs.addClass( 'hide' );
+
+		if ( 'pie' === chart_type ) {
+			$chart_meta_box.find( '.row.vertical-axis, .row.horizontal-axis, .row.y-min' ).addClass( 'hide' );
+		}
+	};
+
+	// Generate a PNG image out of a rendered chart
+	m_chart_chartjs_admin.generate_image_from_chart = function( event ) {
+		var $canvas_source = document.getElementById( 'm-chart-' + event.post_id + '-' + event.instance );
+		var $target_canvas = $( '#m-chart-canvas-render-' + event.post_id );
```

Comment:
Perhaps switch this one to a `$( document.getElementById() )` as well?

Reply:
You know this is a funny one that I honestly don't quite understand. When I use `$( document.getElementById() )` for this it doesn't work. It's like the reference is gotten wrong somehow and at the time I didn't have the wherewithal to chase it down but maybe you see something obvious I was unaware of?